### PR TITLE
Fix Preprints api taxonomy [OSF-7352]

### PR DESCRIPTION
**Pull request by alexschiller**
## Purpose

Taxonomy api is broken

## Changes

Changes the query to work with django

## Side effects

None known
## Ticket
[#OSF-7352]
https://openscience.atlassian.net/browse/OSF-7352

### CODE REVIEW

File: api/preprint_providers/views.py [hunk 1]
```diff
@@ -226,7 +226,7 @@ def get_queryset(self):
             #  Calculate this here to only have to do it once.
             allowed_parents = [id_ for sublist in provider.subjects_acceptable for id_ in sublist[0]]
             allows_children = [subs[0][-1] for subs in provider.subjects_acceptable if subs[1]]
-            return [sub for sub in Subject.find(Q('parents', 'eq', parent)) if provider.subjects_acceptable == [] or self.is_valid_subject(allows_children=allows_children, allowed_parents=allowed_parents, sub=sub)]
+            return [sub for sub in Subject.find(Q('parents___id', 'eq', parent)) if provider.subjects_acceptable == [] or self.is_valid_subject(allows_children=allows_children, allowed_parents=allowed_parents, sub=sub)]
```
Comment (mfraezz) [inline api/preprint_providers/views.py]: This would also work with `Subject.objects.filter(parents___id=parent)`, but that's not necessary until phase 2

Comment (alexschiller) [inline api/preprint_providers/views.py]: IIRC I tried doing it that way and it failed. @cwisecarver pointed out the reason, pretty sure it was a rats nest and decided to do it the easier way until phase 2.

Comment (mfraezz) [inline api/preprint_providers/views.py]: You may have only had two underscores? I tried the above and relevant tests passed.

**Comment by cwisecarver** [comment]
glam

**Pull request merged**